src/scroll-timeline-css.js

```diff
@@ -0,0 +1,184 @@
+import { StyleParser, RegexMatcher } from "./scroll-timeline-css-parser";
+
+const parser = new StyleParser();
+
+function initMutationObserver() {
+  const sheetObserver = new MutationObserver((entries) => {
+    for (const entry of entries) {
+      for (const addedNode of entry.addedNodes) {
+        if (addedNode instanceof HTMLStyleElement) {
+          handleStyleTag(addedNode);
+        }
+        if (addedNode instanceof HTMLLinkElement) {
+          handleLinkedStylesheet(addedNode);
+        }
+      }
+    }
+
+    // TODO: Not a good idea to proxy all elements.
+    // Not adding proxy affects tests that there is no animation,
+    // but there is change for 'animation', 'animation-timeline', etc via style
+    // like this: e.style['animation-timeline'] = "none, auto"
+    // document.querySelectorAll("body *").forEach(element => {
+    //   addProxyForElement(element);
+    // });
+  });
+
+  sheetObserver.observe(document.documentElement, {
+    childList: true,
+    subtree: true,
+  });
+
+  function handleStyleTag(el) { // el: HtmlStyleElement
```

Comment:
Prefer using documentation format consistent with the closure compiler.
see https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler

[@param {Type} varname Description](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#param-type-varname-description)